### PR TITLE
Add InstanceConfig override for /partitionAssignment API

### PR DESCRIPTION
**Pull request by zpinto**
Add InstanceConfig override for /partitionAssignment API to allow for simulating different assignment with different InstanceConfigs.

### Issues
NA

### Description

This change will allow callers of the /partitionAssignment API to pass a list of InstanceConfigs to override the existing ones. Callers will be able to do things like change the DOMAIN config to simulate FAULT_DOMAINs with skewed capacity and more.

### Tests

- [x] testComputePartitionAssignmentWaged 
      - Add another part of the test that uses the InstanceConfig inputField to replace HELIX_ENABLED in the Instance's persisted(ZK) InstanceConfig. While this particular use of the InstanceConfig may not be necessary since we already have ActivateInstance and DeactivateInstance fields, it is the easiest way to test that the InstanceConfig override can be passed.

- The following is the result of the "mvn test" command on the appropriate module:
```
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 116.65 s - in org.apache.helix.rest.server.TestResourceAssignmentOptimizerAccessor
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco:0.8.6:report (generate-code-coverage-report) @ helix-rest ---
[INFO] Loading execution data file /Users/zapinto/Documents/git/zpinto/helix/helix-rest/target/jacoco.exec
[INFO] Analyzed bundle 'Apache Helix :: Restful Interface' with 92 classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:03 min
[INFO] Finished at: 2023-08-09T14:34:45-07:00
[INFO] ------------------------------------------------------------------------
```

### Changes that Break Backward Compatibility (Optional)

Preserves backwards compatibility by adding and not removing old inputFields

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Review by xyuanlu — APPROVED** [review]

**Review by desaikomal — APPROVED** [review]
thanks for offline explanation. Essentially we can override any of the instance-config properties (capacity/toplogy).

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java [hunk 1]
```diff
@@ -104,6 +105,8 @@ private static class InputJsonContent {
   }
 
   private static class InstanceChangeMap {
+    @JsonProperty("InstanceConfigs")
+    JsonNode instanceConfigs;
```
Comment (himanshukandwal) [inline helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java:109]: As this is a new field, so will there be a case where client might pass json with out this property?

Should we use [JsonIgnoreProperties.html#ignoreUnknown](http://fasterxml.github.io/jackson-annotations/javadoc/2.9/com/fasterxml/jackson/annotation/JsonIgnoreProperties.html#ignoreUnknown() )

**Review by himanshukandwal — COMMENTED** [review]
Thanks for this change, Zach!

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java (hunk 1 above)
Comment (zpinto) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java:109]: Yes, this case is covered in the unit tests that already existed. JsonProperty does not require that this field be there in InstanceChangeMap.

**Comment by zpinto** [comment]
This PR is ready to be merged

Final Commit Message:
Add InstanceConfigs inputField to allow overriding the existing InstanceConfigs for /partitionAssignment API.

**Pull request merged**